### PR TITLE
Use trivial resizer if only node expansion supported

### PR DESCRIPTION
**Pull request by mlmhl**
If a driver supports node expansion only, use a trivial resizer to just update PV's capacity and PVC's condition, without any meaningful expand operation.

ref: https://github.com/kubernetes-csi/external-resizer/issues/22

### CODE REVIEW

File: pkg/resizer/csi_resizer.go [hunk 1]
```diff
@@ -46,47 +41,49 @@ const (
...
-	if !supports {
-		return nil, fmt.Errorf("CSI driver does not support controller resize")
+	if !supportControllerResize {
+		supportsNodeResize, err := supportsNodeResize(csiClient, timeout)
+		if err != nil {
+			return nil, fmt.Errorf("failed to check if plugin supports node resize: %v", err)
+		}
+		if supportsNodeResize {
+			klog.Info("The CSI driver supports node resize only, using trivial resizer to handle resize requests")
+			return newTrivialResizer(driverName), nil
```
Comment (gnufied) [inline pkg/resizer/csi_resizer.go]: is there a way to get some tests in? There isn't an e2e that tests this as well, so we have this somewhat untested.  We need an e2e or something that tests the scenario.

Comment (mlmhl) [reply pkg/resizer/csi_resizer.go]: OK, I will try to add some unit tests first.

**Comment by mlmhl** [comment]
@gnufied Add some unit tests, PTAL :)

**Comment by gnufied** [comment]
thanks. Waiting for tests to be green though..

**Comment by mlmhl** [comment]
Oh sorry, I forgot to check the CI output, the tests are green now.

**Pull request merged**